webapi/CopilotChat/Controllers/ChatMemoryController.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticKernel.Memory;
+using SemanticKernel.Service.CopilotChat.Options;
+using SemanticKernel.Service.CopilotChat.Skills.ChatSkills;
+using SemanticKernel.Service.CopilotChat.Storage;
+
+namespace SemanticKernel.Service.CopilotChat.Controllers;
+
+/// <summary>
+/// Controller for getting chat session semantic memory data.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ChatMemoryController : ControllerBase
+{
+    private readonly ILogger<ChatMemoryController> _logger;
+
+    private readonly PromptsOptions _promptOptions;
+
+    private readonly ChatSessionRepository _chatSessionRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatMemoryController"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="promptsOptions">The prompts options.</param>
+    /// <param name="chatSessionRepository">The chat session repository.</param>
+    public ChatMemoryController(
+        ILogger<ChatMemoryController> logger,
+        IOptions<PromptsOptions> promptsOptions,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._logger = logger;
+        this._promptOptions = promptsOptions.Value;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Gets the semantic memory for the chat session.
+    /// </summary>
+    /// <param name="semanticTextMemory">The semantic text memory.</param>
+    /// <param name="chatId">The chat id.</param>
+    /// <param name="memoryName">Name of the memory type.</param>
+    [HttpGet]
+    [Route("chatMemory/{chatId:guid}/{memoryName}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> GetSemanticMemoriesAsync(
+        [FromServices] ISemanticTextMemory semanticTextMemory,
+        [FromRoute] string chatId,
+        [FromRoute] string memoryName)
```

Comment:
Is this the memory type (short vs long)? If so, it would be better if it were renamed to memoryType

Reply:
memoryName is used throughout the application. And it's really just a name. In the future we should define an enum for it.